src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -3024,6 +3044,16 @@ def set_up_network_profile(self, mc: ManagedCluster) -> ManagedCluster:
                     enabled=advanced_network_observability
                 )
             )
+
+        fqdn_policy = self.context.get_enable_fqdn_policy()
+        if fqdn_policy is not None:
+            network_profile.advanced_networking = self.models.AdvancedNetworking(  # pylint: disable=no-member
+                security=self.models.AdvancedNetworkingSecurity(  # pylint: disable=no-member
+                    fqdn_policy=self.models.AdvancedNetworkingFQDNPolicy(
+                        enabled=fqdn_policy
+                    )
```

Comment:
Can observability and security be enabled at the same time? If so, you need to rewrite this to make sure it is empty (None/null) before instantiating the AdvancedNetworking model.

Reply:
Thanks, appreciate the pointer!